spec/json_log_formatter_spec.cr

```diff
@@ -1,25 +1,74 @@
 require "./spec_helper"
 
-describe Dexter::Formatters::JsonLogFormatter do
-  it "formats the data as JSON" do
-    data = {
-      my_data: "is great!",
-    }
+describe Dexter::JSONLogFormatter do
+  it "formats the context data as JSON and skips message if empty" do
     io = IO::Memory.new
+    entry = build_entry({my_data: "is great!"}, source: "json-test", severity: :debug)
 
-    Dexter::Formatters::JsonLogFormatter.new(
-      severity: Logger::Severity::INFO,
-      timestamp: timestamp,
-      progname: "",
-      io: io
-    ).format(data)
+    format(entry, io)
 
     io.to_s.chomp.should eq(
-      {severity: "INFO", timestamp: timestamp, my_data: "is great!"}.to_json
+      {severity: "Debug", source: "json-test", timestamp: timestamp, my_data: "is great!"}.to_json
     )
   end
+
+  it "merge the message if present" do
+    io = IO::Memory.new
+    entry = build_entry({my_data: "is great!"}, message: "my message")
+
+    format(entry, io)
+
+    log = JSON.parse(io.to_s.chomp).as_h
+    log["message"].as_s.should eq("my message")
+    log["my_data"].as_s.should eq("is great!")
+  end
+
+  it "merges exception data if present" do
+    io = IO::Memory.new
+    entry = build_entry({my_data: "is great!"}, exception: RuntimeError.new("This is my error"))
+
+    format(entry, io)
+
+    error = JSON.parse(io.to_s.chomp).as_h["error"]
+    error["class"].as_s.should eq("RuntimeError")
+    error["message"].as_s.should eq("This is my error")
+    # Backtace should be nil since error was instantiated manually
+    error["backtrace"].as_nil.should be_nil
```

Comment:
Could you also use `["backtrace"]?` to get nil instead of casting?

Reply:
Unfortunately no :( This is part of the weirdness I was talking to. Log::Context uses something similar to JSON::Any that requires casting to the type you want first.

I want to open an issue because I don't think it is necessary to have this and may havebeen an oversight

Reply:
Bummer.

Reply:
Yeah, it's not `nil` it's `JSON::Any.new(nil)` 😂 I've been bitten by that so many times.